src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -124,12 +108,12 @@ private static bool TryUseSubstring(ReadOnlySpan<string> uniqueStrings, bool ign
                 // actually perform the comparison as case-sensitive even if case-insensitive
                 // was requested, as there's nothing that would compare equally to the substring
                 // other than the substring itself.
-                bool canSwitchIgnoreCaseToCaseSensitive = ignoreCase;
+                bool canSwitchIgnoreCaseToCaseSensitive = true;
```

Comment:
We know that `ignoreCase` was true on [line 105](https://github.com/dotnet/runtime/pull/88516/files?diff=unified&w=1#diff-8dfef3756fbf6ca0aab7668e6abf54ae6c416c60cb1b3bd86455204942f7dbefR105)

Reply:
Submitted as a distinct PR https://github.com/dotnet/runtime/pull/88709/